nest/src/token/pUSD.sol

```diff
@@ -22,115 +31,157 @@ interface IVault {
+    }
 
     // ========== EVENTS ==========
     event VaultChanged(address oldVault, address newVault);
     event Paused(address account);
     event Unpaused(address account);
 
+    // ========== ROLES ==========
+    bytes32 public constant VAULT_ADMIN_ROLE = keccak256("VAULT_ADMIN_ROLE");
+    bytes32 public constant PAUSER_ROLE = keccak256("PAUSER_ROLE");
+
     // ========== MODIFIERS ==========
     modifier whenNotPaused() {
-        require(!paused, "PUSD: paused");
+        require(!_getpUSDStorage().paused, "pUSD: paused");
         _;
     }
 
-    // ========== CONSTRUCTOR & INITIALIZER ==========
-    /// @custom:oz-upgrades-unsafe-allow constructor
+    /**
+     * @notice Prevent the implementation contract from being initialized or reinitialized
+     * @custom:oz-upgrades-unsafe-allow constructor
+     */
     constructor() {
         _disableInitializers();
     }
 
-    function initialize(address _vault, address admin) external initializer {
-        __ERC20_init("", ""); // Empty strings since we override name() and symbol()
-        __AccessControl_init();
-        __UUPSUpgradeable_init();
-
-        vault = IVault(_vault);
-
-        _grantRole(DEFAULT_ADMIN_ROLE, admin);
-        _grantRole(UPGRADER_ROLE, admin);
-        _grantRole(MINTER_ROLE, admin);
-        _grantRole(BURNER_ROLE, admin);
-        _grantRole(VAULT_ADMIN_ROLE, admin);
-        _grantRole(PAUSER_ROLE, admin);
-    }
-
-    // ========== METADATA OVERRIDES ==========
-    function decimals() public pure override returns (uint8) {
-        return 6;
-    }
-
-    function name() public pure override returns (string memory) {
-        return "Plume USD";
-    }
-
-    function symbol() public pure override returns (string memory) {
-        return "pUSD";
+    /**
+     * @notice Initialize pUSD
+     * @param owner Address of the owner of pUSD
+     * @param asset_ Address of the underlying asset
+     * @param vault_ Address of the Boring Vault
+     */
+    function initialize(address owner, IERC20 asset_, address vault_) public initializer {
+        ComponentToken.initialize(owner, "", "", asset_, false, false);
+
+        pUSDStorage storage $ = _getpUSDStorage();
+        $.vault = IVault(vault_);
+        $.tokenName = "Plume USD";
+        $.tokenSymbol = "pUSD";
+        $.tokenDecimals = 6;
+
+        _grantRole(DEFAULT_ADMIN_ROLE, owner);
+        _grantRole(VAULT_ADMIN_ROLE, owner);
+        _grantRole(PAUSER_ROLE, owner);
+        _grantRole(UPGRADER_ROLE, owner);
     }
 
     // ========== ADMIN FUNCTIONS ==========
     function setVault(
         address newVault
     ) external onlyRole(VAULT_ADMIN_ROLE) {
-        address oldVault = address(vault);
-        vault = IVault(newVault);
+        pUSDStorage storage $ = _getpUSDStorage();
+        address oldVault = address($.vault);
+        $.vault = IVault(newVault);
         emit VaultChanged(oldVault, newVault);
     }
 
     function pause() external onlyRole(PAUSER_ROLE) {
```

Comment:
Let's remove pausing from the token because we don't want the main stablecoin to have a centralized failure point

Reply:
pausable removed.